tests/assert_ansible_schema.py

```diff
@@ -0,0 +1,107 @@
+import re
+import sys
+
+import yaml
+
+
+COMMENT_KEYS = set((
+    "platform",
+    "reboot",
+    "strategy",
+    "complexity",
+    "disruption",
+))
+
+TAGS = set((
+    "strategy",
+    "complexity",
+    "disruption",
+    "severity",
+))
+
+
+def make_parser():
+    parser = ap.ArgumentParser()
+    parser.add_argument("input")
+    return parser
+
+
+def validate_comments(fname):
+    caught_comments = set()
+    regex = re.compile(
+        r"# ({keys}) = .*"
+        .format(keys="|".join(COMMENT_KEYS)))
+    with open(fname, "r") as lines:
+        for line in lines:
+            found = re.search(regex, line)
+            if found:
+                caught_comments.add(found.group(1))
+    assert COMMENT_KEYS == caught_comments, (
+        "Did not found key(s) in comments: {keys}"
+        .format(keys=", ".join(COMMENT_KEYS.difference(caught_comments))))
+
+
+def validate_playbook(playbook):
+    assert "name" in playbook, "playbook doesn't have a name"
+    assert "hosts" in playbook, "playbook doesn't have the hosts entry"
+    assert playbook["hosts"] == "@@HOSTS@@", "playbook's hosts is not set to @@HOSTS@@"
+    assert "become" in playbook, "playbook doesn't have a become key"
+    assert playbook["become"], "become in the playbook is not set to a true-ish value"
+    if "vars" in playbook:
+        assert playbook["vars"], "there are no variables under the 'vars' key of the playbook"
+    assert "tasks" in playbook, "there are no tasks in the playbook"
+    assert "name" in playbook["tasks"][0], "The first task doesn't have name"
+    assert "tags" in playbook, "the playbook doesn't have tags"
+
+    caught_tags = set()
+    tag_regex = re.compile(
+        r".*_({keys})"
+        .format(keys="|".join(TAGS)))
+    cce_regex = re.compile(r"CCE-[0-9]+-[0-9]+")
+    for tag in playbook["tags"]:
+        assert "@" not in tag, \
+            "A playbook tag {tag} contains @, which is unexpected.".format(tag=tag)
+
+        found = re.search(tag_regex, tag)
+        if found:
+            caught_tags.add(found.group(1))
+
+        if re.search(cce_regex, tag):
+            caught_tags.add("CCE")
+
+    assert "CCE" in caught_tags, "There is no CCE tag in the playbook"
+    tags_not_caught = TAGS.difference(caught_tags)
+    assert not tags_not_caught, \
+        "Missing tags: {stray}".format(stray=", ".join(tags_not_caught))
+
+
+def validate_yaml(fname):
+    stream = open(fname, "r")
+    data = yaml.load(stream, Loader=yaml.Loader)
+    for playbook in data:
+        validate_playbook(playbook)
+
+
+def validate_input(fname):
+    try:
+        validate_comments(fname)
+        validate_yaml(fname)
+    except AssertionError as err:
+        msg = (
+            "Error processing {fname}: {err}"
+            .format(fname=fname, err=err)
+        )
+        print(msg, file=sys.stderr)
+        return 1
+    return 0
+
+
+if __name__ == "__main__":
+    parser = make_parser()
+    args = parser.parse_args()
```

Comment:
I think we don't need 3 functions for parsing 1 argument.

Reply:
Now there are 3 arguments :-)